code/modules/events/immovable_rod.dm

```diff
@@ -93,8 +101,14 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 		return ..()
 	//Moved more than 10 tiles in 1 move.
 	var/cur_dist = get_dist(src, destination)
-	if((z != z_original) || (loc == destination) || (FLOOR(cur_dist - previous_distance, 1) > 10))
+	if(z != z_original)
 		qdel(src)
+	if(((loc == destination) || (FLOOR(cur_dist - previous_distance, 1) > 10)) && !looping)
+		qdel(src)
+	if(loc == destination && looping)
+		forceMove(beginning)
+		walk_towards(src, destination, 1)
+		previous_distance = get_dist(src, destination)
```

Comment:
don't really like logic here, you are checking loc == destination twice and looping twice 🥴